shared/src/main/scala/kinesis4cats/producer/fs2/FS2Producer.scala

```diff
@@ -153,31 +155,35 @@ abstract class FS2Producer[F[_], PutReq, PutRes](implicit
           x.toNel
             .traverse_ { x =>
               val (records, deferreds) = x.unzip
-              for {
-                _ <- logger.debug(c.context)(
-                  "Received batch to process"
-                )
-                _ <- underlying.put(records).guaranteeCase {
```

Comment:
Without `attempt`, this just guarantees that the callback is executed, but still leads to `fs2.Stream` failure